docs/index.md

```diff
@@ -967,122 +966,85 @@ With the monkey patch disabled, use `render_component` (or  `render_component_to
 <%= render_component Component.new(message: "bar") %>
 ```
 
-### Sidecar assets (experimental)
+### Sidecar CSS (experimental)
 
-It’s possible to include Javascript and CSS alongside components, sometimes called "sidecar" assets or files.
+_Note: This feature is experimental. Breaking changes should be expected without warning._
 
-To use the Webpacker gem to compile sidecar assets located in `app/components`:
+ViewComponent includes experimental support for encapsulated sidecar CSS, locally scoping CSS selectors using CSS Modules.
 
-1. In `config/webpacker.yml`, add `"app/components"` to the `resolved_paths` array (e.g. `resolved_paths: ["app/components"]`).
-2. In the Webpack entry file (often `app/javascript/packs/application.js`), add an import statement to a helper file, and in the helper file, import the components' Javascript:
+To use the experimental feature, include `ViewComponent::Styleable`:
 
-```js
-import "../components"
-```
+`app/components/styleable_component.rb`:
 
-Then, in `app/javascript/components.js`, add:
-
-```js
-function importAll(r) {
-  r.keys().forEach(r)
-}
-
-importAll(require.context("../components", true, /[_\/]component\.js$/))
+```ruby
+class StyleableComponent < ViewComponent::Base
+  include ViewComponent::Styleable
+end
 ```
 
-Any file with the `_component.js` suffix (such as `app/components/widget_component.js`) will be compiled into the Webpack bundle. If that file itself imports another file, for example `app/components/widget_component.css`, it will also be compiled and bundled into Webpack's output stylesheet if Webpack is being used for styles.
-
-#### Encapsulating sidecar assets
+Add a sidecar stylesheet for the component:
 
-Ideally, sidecar Javascript/CSS should not "leak" out of the context of its associated component.
+`app/components/styleable_component.css`:
 
-One approach is to use Web Components, which contain all Javascript functionality, internal markup, and styles within the shadow root of the Web Component.
+```css
+.foo {
+  color: red;
+}
+```
 
-For example:
+Use `styles` to retrieve the scoped selector:
 
-`app/components/comment_component.rb`
+`app/components/styleable_component.rb`:
 
 ```ruby
-class CommentComponent < ViewComponent::Base
-  def initialize(comment:)
-    @comment = comment
-  end
-
-  def commenter
-    @comment.user
-  end
+class StyleableComponent < ViewComponent::Base
+  include ViewComponent::Styleable
 
-  def commenter_name
-    commenter.name
+  def call
+    content_tag(:div, "Hello, World!", class: styles['foo'])
```

Comment:
```suggestion
    content_tag(:div, "Hello, World!", class: styles.foo)
```

Thoughts on using methods instead of hashes? I think API wise, methods are a little easier to use plus it gives us the ability to later "enhance" the style methods with additional functionality. e.g. Finding unused CSS in the test environment